frontend/src/components/CodebaseListSidebar.vue

```diff
@@ -141,7 +141,7 @@ const initializeFilterValues = () => {
   values.endDate = urlParams.get("publishedBefore")
     ? new Date(urlParams.get("publishedBefore")!)
     : null;
-  values.ordering = urlParams.get("ordering") || "-first_published_at";
+  values.ordering = urlParams.get("ordering") || "relevance";
```

Comment:
Can you explain what behavior this is attempting to fix? The initial sort order should be newest to oldest as "relevance" doesn't make sense on initial load; won't this set the initial sort order to "relevance" when the ordering query param isn't explicitly set?

Reply:
If the user sets the ordering explicitely, the ordering will be available through `urlParams.get("ordering")`. If not, the default ordering during searching (seqrch query is not empty) is "relevance".

But I guess, you also have a point, namely: when there is no search query, just filtering by programming languages. In which case it should be something like:

```
values.ordering = urlParams.get("ordering") || (urlParams.get("query") ? "relevance" : "-first_published_at");
```